### ISSUE TITLE
Issue during debug of tensorflow

### ISSUE THREAD

**Issue by EnricoBeltramo** [tensorflow/models#1021]
Hello,
I'm pretty new to tensorflow and python. If I try to run syntaxnet by command line it works well, but I have an issue when I try to debug it step by step with Eclipse or Spyder. I had same error: an issue on load a library (parser_ops.so). Someone should help me to understand what is the issue? 
I attach here the error log (generated by Spyder)

  File "<ipython-input-2-c92123615836>", line 1, in <module>
    debugfile('/home/erber/models/syntaxnet/bazel-bin/syntaxnet/parser_eval.runfiles/__main__/syntaxnet/parser_eval.py', wdir='/home/erber/models/syntaxnet/bazel-bin/syntaxnet/parser_eval.runfiles/__main__/syntaxnet')

  File "/usr/lib/python2.7/dist-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 719, in debugfile
    debugger.run("runfile(%r, args=%r, wdir=%r)" % (filename, args, wdir))

  File "/usr/lib/python2.7/bdb.py", line 400, in run
    exec cmd in globals, locals

  File "<string>", line 1, in <module>

  File "/usr/lib/python2.7/dist-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 699, in runfile
    execfile(filename, namespace)

  File "/usr/lib/python2.7/dist-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 81, in execfile
    builtins.execfile(filename, *where)

  File "/home/erber/models/syntaxnet/bazel-bin/syntaxnet/parser_eval.runfiles/__main__/syntaxnet/parser_eval.py", line 31, in <module>
    import graph_builder

  File "graph_builder.py", line 22, in <module>
    import load_parser_ops

  File "load_parser_ops.py", line 25, in <module>
    'parser_ops.so'))

  File "/home/erber/.local/lib/python2.7/site-packages/tensorflow/python/framework/load_library.py", line 64, in load_op_library
    None, None, error_msg, error_code)

NotFoundError: parser_ops.so: cannot open shared object file: No such file or directory

thank you very much for any help!

**Comment by aselle** [comment]
@caisq, could you take a look?

**Comment by caisq** [comment]
@aselle This is not directly related to tfdbg (TensorFlow's Debugger), @EnricoBeltramo is using a general-purpose python debugger (in Eclipse/Spyder).

But from the the error stack, I can see that the syntax_net model is trying to load a customly defined op, namely parser_ops. "parser_ops.so" doesn't appear to be a part of TensorFlow. It seems to come from syntaxnet: https://github.com/tensorflow/models/blob/a9133ae914b44602c5f26afbbd7dd794ff9c6637/syntaxnet/syntaxnet/load_parser_ops.py#L23

So further diagnostic questions for @EnricoBeltramo 
1) Does this error happen when you are running the code in non-debug mode?
2) Did you check your LD_LIBRARY_PATH to see if the path to parser_ops.so is included?

**Comment by EnricoBeltramo** [comment]
Yes, exactly: I'm working on Syntaxnet model.  I want to work on it in order to integrate it with other NLP features.
The model works fine if I run demo.sh or parse.sh and parse correctly any phrases as expected. Starting from those demo scripts, I followed the called phyton modules in order to debug it, but when I try to do it by a debugger I have this error.  I tried to configure LD_LIBRARY_PATH with the path of parser_ops.so, but the issue still present. Should be necessary to configure some other library path/environment variables in order to launch from debug?

Thank you
